test/distributed/rpc/faulty_agent/test_faulty_agent_rpc_trap.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import sys
+
+# Buck-specific import to call into real test file
+import __test_main__
+
+import torch.testing._internal.faulty_rpc_agent.rpc  # noqa
+from torch.testing._internal import dist_utils
+import torch.distributed.rpc as rpc
+
+# The following message types are currently retried in the RREF protocol and
+# distributed autograd. Thus only these messages should be tested with the
+# Faulty RPC Agent.
+retryable_message_types = ["RREF_FORK_REQUEST",
+                           "RREF_CHILD_ACCEPT",
+                           "RREF_USER_DELETE",
+                           "CLEANUP_AUTOGRAD_CONTEXT_REQ"]
+
+# Configure TEST_CONFIG.
+dist_utils.TEST_CONFIG.rpc_backend_name = "FAULTY_PROCESS_GROUP"
+dist_utils.TEST_CONFIG.build_rpc_backend_options = lambda test_object: rpc.backend_registry.construct_rpc_backend_options(
+    test_object.rpc_backend,
+    init_method=test_object.init_method,
+    num_send_recv_threads=8,
+    num_fail_sends=1,
```

Comment:
Can we make this higher (like 3) to ensure better test coverage? If it increases test execution time dramatically then lets leave it as it is.